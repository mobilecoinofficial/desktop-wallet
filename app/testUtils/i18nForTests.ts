import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import enUS from '../../locales/enUS/translation.json';

i18n.use(initReactI18next);
if (!i18n.isInitialized) {
  i18n.init({
    lng: 'enUS',
    resources: {
      enUS,
    },
    supportedLngs: ['enUS'],
  });
}

export default i18n;
