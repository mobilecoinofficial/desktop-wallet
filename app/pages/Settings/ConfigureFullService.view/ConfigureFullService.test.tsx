import React from 'react';

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import '../../../testUtils/i18nForTests';

import { ConfigureFullServiceView } from './ConfigureFullService.view';

describe('ConfigureFullServiceView', () => {
  const mockOnClickBack = jest.fn();
  const mockToggle = jest.fn();
  const selectedAccount = {
    account: {
      accountId: '1234',
      firstBlockIndex: '1',
      key_derivation_version: '1',
      mainAddress: 'b58 code',
      name: null,
      nextSubaddressIndex: '1235',
      object: 'account',
    },
    balanceStatus: {
      accountBlockIndex: '1234',
      isSynced: true,
      localBlockIndex: '1234',
      networkBlockIndex: '1234',
      object: 'balance',
      orphanedPmob: '1234',
      pendingPmob: '1234',
      secretedPmob: '1234',
      spentPmob: '1234',
      unspentPmob: '1234',
    },
  };
  const configureFullServiceConfigs = {
    fullServiceDbPath: 'fullService/path',
    leaveFullServiceRunning: false,
    ledgerDbPath: 'ledger/path',
    toggleLeaveFullServiceRunning: mockToggle,
  };

  describe('component', () => {
    describe('render', () => {
      test('ConfigureFullServiceView renders correctly', () => {
        const { asFragment } = render(
          <ConfigureFullServiceView
            onClickBack={mockOnClickBack}
            selectedAccount={selectedAccount}
            configureFullServiceConfigs={configureFullServiceConfigs}
          />
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('breadcrumb navigation', () => {
      test('Settings breadcrumb navigates away from current view', () => {
        const { getByText } = render(
          <ConfigureFullServiceView
            onClickBack={mockOnClickBack}
            selectedAccount={selectedAccount}
            configureFullServiceConfigs={configureFullServiceConfigs}
          />
        );

        const configMessage =
          'This screen shows how to customize the behavior of background ledger processor (named Full-Service) as well as how to perform some basic resets.';
        expect(getByText(configMessage)).toBeInTheDocument();
        userEvent.click(screen.getByText('Settings'));
        expect(mockOnClickBack).toHaveBeenCalled();
      });
    });
  });
});
