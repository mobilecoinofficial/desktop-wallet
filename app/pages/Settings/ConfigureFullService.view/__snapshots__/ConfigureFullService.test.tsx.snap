// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureFullServiceView component render ConfigureFullServiceView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-cardContainer-1 MuiContainer-maxWidthSm"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <ol
        class="MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <button
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorInherit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Settings
            </p>
          </button>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          &gt;
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            Manage Ledger Database
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayInline"
      >
        This screen shows how to customize the behavior of background ledger processor (named Full-Service) as well as how to perform some basic resets.
      </p>
      <div
        class="MuiBox-root MuiBox-root-6"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Full-Service is a standalone executable which provides blockchain synchronization and wallet services.
      </p>
      <div
        class="MuiBox-root MuiBox-root-7"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        It creates encrypted, attested connections to validator nodes that participate in federated voting for agreements on current block height, block headers, and to submitting transactions. These validator nodes are considered highly trusted due to their use of SGX, and they are used as a reliable source for block information, as well as for validating and processing the proposed transactions from Full-Service.
      </p>
    </div>
    <div
      class="MuiBox-root MuiBox-root-8"
    >
      <div
        class="MuiBox-root MuiBox-root-9"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Ledger Status
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayInline"
        >
          The ledger is synced.
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root-11"
      >
        <div
          class="MuiTableContainer-root"
        >
          <table
            aria-label="block status"
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Block Type
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Current Height
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Maximum Height
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                  scope="col"
                >
                  % Synced
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                  role="cell"
                  scope="row"
                >
                  Network Blocks
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  1234
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                />
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                  role="cell"
                  scope="row"
                >
                  Local Blocks
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  1234
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  1234
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  100
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                  role="cell"
                  scope="row"
                >
                  Account Blocks
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  1234
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  1234
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  100
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Network Blocks represent the total blocks within the MobileCoin Network. This is the value your Local Blocks and Account Blocks must equal the total amount of Network Blocks to be synced 100 percent.
      </p>
      <div
        class="MuiBox-root MuiBox-root-12"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Local Blocks shows the blocks your Full-Service has downloaded from ledger.
      </p>
      <div
        class="MuiBox-root MuiBox-root-13"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Account Blocks shows the blocks that a specific Account has synced. Each account has its own Account Blocks sync. Once this value is synced, you will immediately see the effects of transactions.
      </p>
    </div>
    <div
      class="MuiBox-root MuiBox-root-14"
    >
      <div
        class="MuiBox-root MuiBox-root-15"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Full-Service Background Process
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root-16"
      >
        <div
          class="MuiBox-root MuiBox-root-17"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            Leave Full-Service Active is off
          </p>
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-22 MuiSwitch-input"
                    name="checkedC"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayInline"
        >
          Full-Service will not continue to sync when you exit the wallet.
        </p>
        <div
          class="MuiBox-root MuiBox-root-23"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          By default, Full-Service terminates when you exit the wallet. This setting allows you to leave Full-Service syncing as a background process. It is best to keep the ledger synced to avoid delays in transactions or balance fetches.
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-24"
    >
      <div
        class="MuiBox-root MuiBox-root-25"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Full-Service Directory
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root-26"
      >
        <div
          class="MuiBox-root MuiBox-root-27"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Ledger Database Path:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          ledger/path
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Export
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Import
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root MuiBox-root-28"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Full-Service Database Path:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          fullService/path
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Export Tx History
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root MuiBox-root-29"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Full-Service Binaries Path:
        </p>
        <div
          style="cursor: pointer;"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            undefined
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
